### PR TITLE
support `--host` option for `daml deploy` command

### PR DESCRIPTION
**Pull request by nickchapman-da**
- support `--host` option for `daml deploy` command

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: language-support/hs/bindings/src/DA/Ledger.hs [hunk 1]
```diff
@@ -34,8 +35,11 @@ import DA.Ledger.Types
 import UnliftIO (liftIO,timeout,bracket)
 
 configOfPort :: Port -> ClientConfig
-configOfPort port =
-    ClientConfig { clientServerHost = Host "localhost"
+configOfPort = configOfHostAndPort "localhost"
+
+configOfHostAndPort :: Host -> Port -> ClientConfig
+configOfHostAndPort host port =
```
Comment (cocreature) [inline language-support/hs/bindings/src/DA/Ledger.hs]: I would probably just inline the definition at this point. The helper doesn’t add that much. But I don’t feel strongly about this so your choice.

File: daml-assistant/daml-helper/src/DamlHelper/Run.hs [hunk 2]
```diff
@@ -671,16 +672,16 @@ runStart sandboxPort (StartNavigator shouldStartNavigator) (OpenBrowser shouldOp
...
     darPath <- getDarPath
     doBuild
     let SandboxPort port = sandboxPort
-    putStrLn $ "Deploying " <> darPath <> " to ledger on port " <> show port
+    putStrLn $ "Deploying " <> darPath <> " to ledger on " <> host <> ":" <> show port
     bytes <- BS.readFile darPath
     let ls = L.uploadDarFile bytes
     let timeout = 30 :: L.TimeoutSeconds
-    let ledgerClientConfig = L.configOfPort $ L.Port port
+    let ledgerClientConfig = L.configOfHostAndPort (L.Host $ fromString host) (L.Port port)
```
Comment (cocreature) [inline daml-assistant/daml-helper/src/DamlHelper/Run.hs]:
```suggestion
    let ledgerClientConfig = L.configOfHostAndPort (fromString host) (L.Port port)
```

File: daml-assistant/daml-helper/src/DamlHelper/Main.hs [hunk 3]
```diff
@@ -35,7 +35,7 @@ data Command
     | New { targetFolder :: FilePath, templateNameM :: Maybe String }
     | Migrate { targetFolder :: FilePath, mainPath :: FilePath, pkgPathFrom :: FilePath, pkgPathTo :: FilePath }
     | Init { targetFolderM :: Maybe FilePath }
-    | Deploy { optSandboxPort :: Maybe SandboxPort }
+    | Deploy { optHostname :: Maybe String, optSandboxPort :: Maybe SandboxPort }
```
Comment (hurryabit) [inline daml-assistant/daml-helper/src/DamlHelper/Main.hs]: Is this the host where the sandbox is running? If so, why not `optSandboxHost` for the sake of consistency?

Comment (nickchapman-da) [reply daml-assistant/daml-helper/src/DamlHelper/Main.hs]: We probably want to move the other direction, and talks about Ledgers, not Sandbox.
This was some feedback we got from users of the `deploy` command.

**Pull request merged**